### ISSUE TITLE
Notify x minutes before scheduled time

### ISSUE THREAD

**Issue by jpgichw** [orgzly-revived/orgzly-android-revived#171]
- I have searched for existing issues that may be the same as or related to mine.

in org-wild-notifier package for emacs it has a feature where you can set multiple alert times before (including 0 minutes before alerting)

it could be set in a configuration file or could be overridden with an org property:

```
  (setq org-wild-notifier-alert-time '(10 0))
```

```
* TODO Visit the chocolate factory
  SCHEDULED: <2018-01-04 Thu 16:00>
  :PROPERTIES:
  :WILD_NOTIFIER_NOTIFY_BEFORE: 80 60 55 43 5
  :END:
```

something similiar to this would be nice, maybe put the config in the reminders settings and in the edit timestamps window for adding the property

additionally, the property should be the same as org-wild-notifier. since most people use that in emacs and might have orgzly on mobile as well.

**Comment by jpgichw** [comment]
maybe the property should be :NOTIFY_BEFORE: actually, but if that's the case org-wild-notifier should use that property as well.

cc @akhramov @bcc32 @colonelpanic8

**Comment by Ypot** [comment]
Should it have RFC 5545 format, to be compatible with ICS calendars?


Example:  The following example is for a "VALARM" calendar component
      that specifies an audio alarm that will sound at a precise time
      and repeat 4 more times at 15-minute intervals:

       BEGIN:VALARM
       TRIGGER;VALUE=DATE-TIME:19970317T133000Z
       REPEAT:4
       DURATION:PT15M
       ACTION:AUDIO
       ATTACH;FMTTYPE=audio/basic:ftp://example.com/pub/
        sounds/bell-01.aud
       END:VALARM

      The following example is for a "VALARM" calendar component that
      specifies a display alarm that will trigger 30 minutes before the
      scheduled start of the event or of the to-do it is associated with
      and will repeat 2 more times at 15-minute intervals:









Desruisseaux                Standards Track                    [Page 75]

RFC 5545                       iCalendar                  September 2009


       BEGIN:VALARM
       TRIGGER:-PT30M
       REPEAT:2
       DURATION:PT15M
       ACTION:DISPLAY
       DESCRIPTION:Breakfast meeting with executive\n
        team at 8:30 AM EST.
       END:VALARM

      The following example is for a "VALARM" calendar component that
      specifies an email alarm that will trigger 2 days before the
      scheduled due DATE-TIME of a to-do with which it is associated.
      It does not repeat.  The email has a subject, body, and attachment
      link.

       BEGIN:VALARM
       TRIGGER;RELATED=END:-P2D
       ACTION:EMAIL
       ATTENDEE:mailto:john_doe@example.com
       SUMMARY:*** REMINDER: SEND AGENDA FOR WEEKLY STAFF MEETING ***
       DESCRIPTION:A draft agenda needs to be sent out to the attendees
         to the weekly managers meeting (MGR-LIST). Attached is a
         pointer the document template for the agenda file.
       ATTACH;FMTTYPE=application/msword:http://example.com/
        templates/agenda.doc
       END:VALARM

Page 75
https://www.rfc-editor.org/rfc/rfc5545#section-3.6.6

**Comment by bcc32** [comment]
> maybe the property should be :NOTIFY_BEFORE: actually, but if that's the case org-wild-notifier should use that property as well.
> 
> cc @akhramov @bcc32 @colonelpanic8

I think it would be fine for org-wild-notifier to check both properties and use whichever exists (preferring the specific one if both exist, I guess).  But I don't plan on proposing it there until a decision is made here.

**Comment by pjhfggij** [comment]
That would be very very useful. Couldn't notifications though the `SCHEDULED` property (added via the "delay" option in orgzly)? Although I also use org wild notifier on the desktop, so if orgzly could integrate with it that would be even better.

**Comment by alensiljak** [comment]
What is the purpose of this? Is it difficult to move the scheduled time by whatever value goes into this field?

**Comment by pjhfggij** [comment]
@alensiljak I think it would be cleaner. You have a meeting at 3pm, but you need to get ready 15 minutes before and maybe prepare something an hour before. Right now, I am copying item for each one of these times with different times, but it's a bit cumbersome.

**Comment by pjhfggij** [comment]
@alensiljak I don't mind orgzly creating three separate items if that's a preferred format, but I think it's easier if it lets you do that directly in the "Schedule time" dialog rather than having to go out of this activity, holding to select a newly created item, selecting copy,  holding to select a newly created item, selecting paste, opening, adjusting scheduled date (and repeating for each time)

I personally think it would be cleaner if multiple scheduled times were in a single header to avoid duplication, but that's just an opinion

**Comment by alensiljak** [comment]
I agree that having an option to create multiple notifications, just by adding another timestamp in the header would be a great option. That idea would be worth elaborating on, probably.
In that case the title is a bit misleading.
